### PR TITLE
Refactor Performance Report code to fix doomsday bugs

### PR DESCRIPTION
**Pull request by Dantemss**
- Order of tasks should now match the table headings
- Tasks ordered by due date
- Students ordered by last name
- Average now divides by all problems in the homework, rather than just completed problems
- Missing tasks (period change etc) are now sent as nils
- `PerformanceReportExport` now returns the uploaded file path instead of the tempfile path
- Cleaned up Performance Report specs that left temp files behind

### CODE REVIEW

**Comment by jpslav** [comment]
Looks good to me.

File: app/representers/api/v1/performance_report_representer.rb [hunk 1]
```diff
@@ -3,6 +3,20 @@ class PerformanceReportRepresenter < Roar::Decorator
...
+        nil
+      end
+
+      def to_json(*args)
+        'null'
+      end
+
+    end
+
     class StudentData < Roar::Decorator
```
Comment (nathanstitt) [inline app/representers/api/v1/performance_report_representer.rb]: @Dantemss I think maybe you mis-understood the "null" I suggested you send the FE.  I mean that the property for the student would be missing, not a string "null". This works though, so if you're happy with it we can roll this way.

Comment (Dantemss) [inline app/representers/api/v1/performance_report_representer.rb]: It comes out as a javascript/json null, I believe (no quotes).

Comment (nathanstitt) [inline app/representers/api/v1/performance_report_representer.rb]: Oh I see.  Cool, I thought you were returning the string "null".  Thanks for explaining :)

**Pull request merged**